modules/machineconfig-modify-dns-nameservers.adoc

```diff
@@ -0,0 +1,174 @@
+// Module included in the following assemblies:
+//
+// * post_installation_configuration/machine-configuration-tasks.adoc
+
+[id="machineconfig-modify-dns-nameservers_{context}"]
+= Modifying the DNS name servers
+
+You can override the DNS name servers and search list in the `/etc/resolv.conf` file.
+Some operating systems use `NetworkManager` to manage network interfaces.
+When `NetworkManager` manages some interfaces, it also manages the contents of the `/etc/resolv.conf` file.
+The following procedure shows how to replace the `/etc/resolv.conf` file and prevent `NetworkManager` from managing the file.
+
+.Procedure
+
+. Create the contents of the `resolv.conf` file and encode it as base64. For example:
++
+[source,terminal]
+----
+$ cat << EOF | base64
+# Custom resolv.conf added with Machine Config Operator
+# and the override-etc-resolv-conf.yaml file.
+search svc.cluster.local cluster.local example.com
+nameserver 192.168.24.1
+nameserver 10.0.0.1
+nameserver 10.0.0.2
+EOF
+----
++
+.Example output
+[source,terminal]
+----
+IyBDdXN0b20gcmVzb2x2LmNvbmYgYWRkZWQgd2l0aCBNYWNoaW5lIENvbmZpZyBPcGVyYXRvcgoj
+IGFuZCB0aGUgb3ZlcnJpZGUtZXRjLXJlc29sdi1jb25mLnlhbWwgZmlsZS4Kc2VhcmNoIHN2Yy5j
+bHVzdGVyLmxvY2FsIGNsdXN0ZXIubG9jYWwgZXhhbXBsZS5jb20KbmFtZXNlcnZlciAxOTIuMTY4
+LjI0LjEKbmFtZXNlcnZlciAxMC4wLjAuMQpuYW1lc2VydmVyIDEwLjAuMC4yCg==
+----
+
+. Create the contents of a `disable-network-manager-dns.conf` file to prevent `NetworkManager` from managing the `/etc/resolv.conf` file:
++
+[source,terminal]
+----
+$ cat << EOF | base64
+[main]
+dns=none
+EOF
+----
++
+.Example output
+[source,terminal]
+----
+W21haW5dCmRucz1ub25lCg==
+----
+
+. Create the `MachineConfig` object file. Replace the base64 strings with the values from the previous steps.
+This example adds the machine configuration to the nodes with the `master` role.
+You can change it to `master` or `worker` or make a separate `MachineConfig` object for the each role:
++
+[source,terminal]
+----
+$ cat << EOF > ./override-etc-resolv-conf.yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: override-etc-resolv-conf
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64, IyBDdXN0b20gcmVzb2x2LmNvbmYgYWRkZWQgd2l0aCBNYWNoaW5lIENvbmZpZyBPcGVyYXRvcgojIGFuZCB0aGUgb3ZlcnJpZGUtZXRjLXJlc29sdi1jb25mLnlhbWwgZmlsZS4Kc2VhcmNoIHN2Yy5jbHVzdGVyLmxvY2FsIGNsdXN0ZXIubG9jYWwgZXhhbXBsZS5jb20KbmFtZXNlcnZlciAxOTIuMTY4LjI0LjEKbmFtZXNlcnZlciAxMC4wLjAuMQpuYW1lc2VydmVyIDEwLjAuMC4yCg==
```

Comment:
In this case, the user definitely needs to replace this with their own base64 output, or the outcome is probably not going to be what is intended.